### PR TITLE
cgen: Fix struct fixedarray init

### LINKED ISSUE #16690: cgen error when using a fixed array, for initialising a sumtype field
### Describe the bug

A cgen error when compiling:
```v
type Data = int | [3]int

struct Sample {
    data Data
}

test := Sample{
    data: [5, 10, 15]!
    // data: 5 /* works fine */
}

println(test)
```

### Expected Behavior

The code should compile and run cleanly, showing something similar to this:
```
Sample{
    data: Data([5, 10, 15]!)
}
```

### Current Behavior

Cgen error:
```
#0 11:22:01 ᛋ master /v/vnew❱v run bug.v 
==================
/tmp/v_1000/bug.3616875129465192796.tmp.c:12204: error: ';' expected (got "}")
...
==================
(Use `v -cg` to print the entire error message)
builder error: 
==================
C error. This should never happen.
This is a compiler bug, please report it using `v bug file.v`.
https://github.com/vlang/v/issues/new/choose
You can also use #help on Discord: https://discord.gg/vlang
#1 11:22:03 ᛋ master /v/vnew❱
```

### Reproduction Steps

Try to compile the code from the description with latest V 0.3.2 f0a252d.

### Possible Solution

Fixing cgen, and may be the checker too.

### Additional Information/Context

The following workaround works (wrapping the fixed array in a struct):
```v
struct Fixed { value [3]int }
type Data = int | Fixed
struct Sample { data Data }
test := Sample{
        data: Fixed{ [5, 10, 15]! }
}
println(test)
```
... producing:
```
Sample{
    data: Data(Fixed{
        value: [5, 10, 15]
    })
}
```

### V version

V 0.3.2 f0a252d

### Environment details (OS name and version, etc.)

```
OS: linux, Ubuntu 20.04.5 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz
CC version: cc (Ubuntu 10.3.0-1ubuntu1~20.04) 10.3.0

getwd: /v/vnew
vmodules: /home/delian/.vmodules
vroot: /v/vnew
vexe: /v/vnew/v
vexe mtime: 2022-12-16 09:23:03
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.2 3643785.f0a252d

Git version: git version 2.39.0
Git vroot status: weekly.2022.50-24-gf0a252d0
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3
```

### PR DESCRIPTION
**Pull request by felipensp**
Fix #16690

```V
type Data = int | [3]int

struct Sample {
    data Data
}

test := Sample{
    data: [5, 10, 15]!
    //data: 5 /* works fine */
}

println(test)
println(test.data)
```

### CODE REVIEW

**Comment by Delta456** [comment]
Can you add tests as well

**Comment by felipensp** [comment]
> Can you add tests as well

Sure, I'll add it. Just checking if it is breaking something. 👍🏻

File: vlib/v/checker/tests/struct_fixed_array_init_test.vv [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+fn test_main() {
+	
+
+	test := Sample{
+		data: [5, 10, 15]!
+		//data: 5 /* works fine */
+	}
+
+	println(test)
+	println(test.data)
```
Comment (spytheman) [inline vlib/v/checker/tests/struct_fixed_array_init_test.vv:17]:
```suggestion
	println(test.data)
	assert test.data.str() == 'Data([5, 10, 15])'
```

**Review by spytheman — APPROVED** [review]
Excellent work.

**Pull request merged**